main.py

```diff
@@ -0,0 +1,111 @@
+        followers = self.api.getTotalFollowers(user_id)
+        return [follower['username'] for follower in followers]
+
+    def get_followings(self, user_id):
+        """Returns list of user's followings."""
+        followings = self.api.getTotalFollowings(user_id)
+        return [following['username'] for following in followings]
+
+    def create_database(self):
+        engine = create_engine('sqlite:///users.db')
+        Base.metadata.create_all(engine)
+        Session = sessionmaker(bind=engine)
+        return Session()
+
+    def time_now(self):
+        return datetime.datetime.now().strftime('%Y-%m-%d %H:%M')
+
+    def save_db(self, connect_database, followers, followings):
+        """Written followers and followings user to database"""
+        session = connect_database
+        now = self.time_now()
+        for (followings, followers) in itertools.zip_longest(followings, followers, fillvalue=''):
+            text = InstagramUsers(followings=followings, followers=followers, status="new", time=str(now))
+            session.add(text)
+        session.commit()
+
+    def database_followings(self, connect_database):
+        """Getting all followings of the database"""
+        session = connect_database
+        database_followings = session.query(InstagramUsers.followings).all()
+        return [y for x in database_followings for y in x]
+
+    def database_follower(self, connect_database):
+        """Getting all followers of the database"""
+        session = connect_database
+        database_follower = session.query(InstagramUsers.followers).all()
+        return [y for x in database_follower for y in x]
+
+    def subscribed(self, connect_database, new_followers, new_followings):
+        """Check whether the current user is in the database.If not, then add(status- subscribed)"""
+        session = connect_database
+        now = self.time_now()
+        for following in new_followings:  # write to database subscribed followings
+            if following not in self.database_followings(connect_database):
+                text = InstagramUsers(followings=following, followers='', status="subscribed", time=str(now))
+                session.add(text)
+            session.commit()
+        for follower in new_followers:  # write to database subscribed followers
+            if follower not in self.database_follower(connect_database):
+                text = InstagramUsers(followings='', followers=follower, status="subscribed", time=str(now))
+                session.add(text)
+            session.commit()
+
+    def unsubscribed(self, connect_database, new_followers, new_followings):
+        """Check whether the current database user is in the current list.If not, then add(status-unsubscribed)"""
+        session = connect_database
+        now = self.time_now()
+
+        for following_l in self.database_followings(connect_database):  # write to database unsubscribed followings
```

Comment:
This block of code looks similar to the other 4 same blocks.
I guess it might be simplified using an additional function.

Reply:
in process